symengine/polynomial.h

```diff
@@ -8,9 +8,253 @@
 #include <symengine/expression.h>
 #include <symengine/monomials.h>
 
+// Calculates bit length of number, used in UIntDict*= only
+template <typename T>
+unsigned int bit_length(T t)
+{
+    unsigned int count = 0;
+    while (t > 0) {
+        count++;
+        t = t >> 1;
+    }
+    return count;
+}
+
 namespace SymEngine
 {
 
+class UIntDict
```

Comment:
This class and `UnivariateExprPolynomial` differs only with the multiplication algorithm right? Isn't it possible to have shared code for both?